README.md

```diff
@@ -166,349 +48,49 @@ Preact supports modern browsers and IE9+:
-
-
----
-
-
-## Linked State
-
-One area Preact takes a little further than React is in optimizing state changes. A common pattern in ES2015 React code is to use Arrow functions within a `render()` method in order to update state in response to events.  Creating functions enclosed in a scope on every render is inefficient and forces the garbage collector to do more work than is necessary.
-
-One solution to this is to bind component methods declaratively.
-Here is an example using [decko](http://git.io/decko):
-
-```js
-class Foo extends Component {
-	@bind
-	updateText(e) {
-		this.setState({ text: e.target.value });
-	}
-	render({ }, { text }) {
-		return <input value={text} onInput={this.updateText} />;
-	}
-}
-```
-
-While this achieves much better runtime performance, it's still a lot of unnecessary code to wire up state to UI.
-
-Fortunately there is a solution, in the form of a module called [linkstate](https://github.com/developit/linkstate). Calling `linkState(component, 'text')` returns a function that accepts an Event and uses its associated value to update the given property in your component's state. Calls to `linkState()` with the same arguments are cached, so there is no performance penalty.  Here is the previous example rewritten using _Linked State_:
-
-```js
-import linkState from 'linkstate';
-
-class Foo extends Component {
-	render({ }, { text }) {
-		return <input value={text} onInput={linkState(this, 'text')} />;
-	}
-}
-```
-
-Simple and effective. It handles linking state from any input type, or an optional second parameter can be used to explicitly provide a keypath to the new state value.
-
-> **Note:** In Preact 7 and prior, `linkState()` was built right into Component. In 8.0, it was moved to a separate module. You can restore the 7.x behavior by using linkstate as a polyfill - see [the linkstate docs](https://github.com/developit/linkstate#usage).
-
-
-
-## Examples
-
-Here is a somewhat verbose Preact `<Link>` component:
-
-```js
-class Link extends Component {
-	render(props, state) {
-		return <a href={props.href}>{props.children}</a>;
-	}
-}
-```
-
-Since this is ES6/ES2015, we can further simplify:
-
-```js
-class Link extends Component {
-    render({ href, children }) {
-        return <a {...{ href, children }} />;
-    }
-}
-
-// or, for wide-open props support:
-class Link extends Component {
-    render(props) {
-        return <a {...props} />;
-    }
-}
+// create our tree:
+render(<main><h1>Hello</h1></main>, document.body);
 
-// or, as a stateless functional component:
-const Link = ({ children, ...props }) => (
-    <a {...props}>{ children }</a>
-);
+// update the tree in-place:
+render(<main><h1>Hello World!</h1></main>, document.body);
+// ^ only updates the text of our <h1>
```

Comment:
// ^ this second invocation of `render()` will only use a single DOM call to update the text of the <h1>